controls/V-41614.rb

```diff
@@ -1,54 +1,76 @@
+  within the web server, investigation into the cause of event is severely
+  hindered. Log record content that may be necessary to satisfy the requirement
+  of this control includes, but is not limited to, time stamps, source and
+  destination IP addresses, user/process identifiers, event descriptions,
+  application-specific events, success/fail indications, file names involved,
+  access control, or flow control rules invoked.
+  "
+  desc  "check", "Review the NGINX web server documentation and deployment configuration 
+  to determine if the web server is configured to generate sufficient information to resolve 
+  in which process within the web server the log event occurred.
+
+  Check for the following:
+    # grep the 'log_format' directive in the http context of the nginx.conf. 
+
+  The logs will not include sufficient information if the 'log_format' directive does not exist.
+
+  If the the 'log_format' directive does not exist, this is a finding.
+
+  Example configuration:
+  log_format  main  '$remote_addr - $remote_user [$time_local] ""$request""'
+  '$status $body_bytes_sent ""$http_referer""'
+  '""$http_user_agent"" ""$http_x_forwarded_for""';
   "
+  desc  "fix", "  
+  Configure the 'log_format' directive in the nginx.conf file to look like the following:
+
+  log_format  main  '$remote_addr - $remote_user [$time_local] ""$request""'
+  '$status $body_bytes_sent ""$http_referer""'
+  '""$http_user_agent"" ""$http_x_forwarded_for""';
+
+  NOTE: Your log format may be using different variables based on the determination of what 
+  information is sufficient in order to establish where the event occured."
+
   impact 0.5
+  tag "severity": "medium"
   tag "gtitle": "SRG-APP-000097-WSR-000058"
   tag "gid": "V-41614"
   tag "rid": "SV-54191r3_rule"
   tag "stig_id": "SRG-APP-000097-WSR-000058"
   tag "fix_id": "F-47073r2_fix"
   tag "cci": ["CCI-000132"]
   tag "nist": ["AU-3", "Rev_4"]
-  tag "false_negatives": nil
-  tag "false_positives": nil
-  tag "documentable": false
-  tag "mitigations": nil
-  tag "severity_override_guidance": false
-  tag "potential_impacts": nil
-  tag "third_party_tools": nil
-  tag "mitigation_controls": nil
-  tag "responsibility": nil
-  tag "ia_controls": nil
-  tag "check": "Review the web server documentation and deployment
-configuration to determine if the web server is configured to generate
-sufficient information to resolve in which process within the web server the
-log event occurred.
-
-Request a user access the hosted application and generate logable events, and
-then review the logs to determine if the process of the event within the web
-server can be established.
-
-If it cannot be determined where the event occurred, this is a finding."
-  tag "fix": "Configure the web server to generate enough information to
-determine in what process within the web server the log event occurred."
+
+  nginx_conf_handle = nginx_conf(input('conf_path'))
```

Comment:
I don't understand this actually - why is the nginx_conf resource being used here?